### DISCUSSION TITLE
cascade layer order

### DISCUSSION THREAD

**Discussion by glRajkumar2** [tailwindlabs/tailwindcss#10520] [Help]
Following is my css file setup,

```css
@tailwind` base;
@tailwind components;
@tailwind utilities;

@layer base {
  button:not(.not-theme) {
    @apply w-fit px-4 py-1;
  }
}
````

And then trying to style button like below 

```jsx
<button className="px-8">My btn</button>
```

But utility style (px-8) overridden by base layer default style.

**Comment by wongjn** [comment]
This is because the selector `button:not(.not-theme)` has a specificity of `0-1-1` which is higher than the specificity of `px-8` which is `0-1-0`. You could change the selector with `:where()` to reduce the specificity in a multitude of ways, such as `button:where(:not(.not-theme))` or `:where(button:not(.not-theme))`.
